src/main/java/com/spaceuptech/api/core/utils/Transport.java

```diff
@@ -0,0 +1,129 @@
+package com.spaceuptech.api.core.utils;
+
+import com.google.protobuf.ByteString;
+import com.spaceuptech.api.core.proto.*;
+import com.spaceuptech.api.core.proto.Response;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutput;
+import java.io.ObjectOutputStream;
+import java.util.concurrent.TimeUnit;
+
+public class Transport {
+
+    public static Meta makeMeta(String project, String collection, String dbType, String token) {
+
+        Meta.Builder metaBuilder = Meta.newBuilder();
+
+        if (project != null) metaBuilder.setProject(project);
+        if (collection != null) metaBuilder.setCol(collection);
+        if (dbType != null) metaBuilder.setDbType(dbType);
+        if (token != null) metaBuilder.setToken(token);
+
+        return metaBuilder.build();
+    }
+
+    private static StreamObserver<Response> makeStreamObserver(Utils.ResponseListener listener, ManagedChannel channel) {
+        return new StreamObserver<Response>() {
+            @Override
+            public void onNext(Response value) {
+
+                System.out.println(value);
+
+//                if (value.getStatus() == 200)
+//                    listener.onResponse(value.getStatus(), new com.spaceuptech.api.core.utils.Response(new JsonParser().parse(value.getResult().toString()).getAsJsonObject()));
+//                else
+//                    listener.onResponse(value.getStatus(), new com.spaceuptech.api.core.utils.Response(new JsonParser().parse(value.getError()).getAsJsonObject()));
+
+                try {
+                    channel.shutdown().awaitTermination(1, TimeUnit.SECONDS);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+
+            @Override
+            public void onError(Throwable t) {
+                listener.onError(new Exception(t));
+            }
+
+            @Override
+            public void onCompleted() {}
+        };
+    }
+
+    public static void create(String host, int port, Object doc, String operation, Meta meta, Utils.ResponseListener listener) {
+
+        final ManagedChannel channel = ManagedChannelBuilder.forAddress(host, port).usePlaintext().build();
+        final SpaceCloudGrpc.SpaceCloudStub stub = SpaceCloudGrpc.newStub(channel);
+
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        ObjectOutput out;
+        try {
+            out = new ObjectOutputStream(bos);
+            out.writeObject(doc);
+            out.flush();
+            byte[] bytes = bos.toByteArray();
+
+            CreateRequest createRequest = CreateRequest.newBuilder()
+                    .setDocument(ByteString.copyFrom(bytes))
```

Comment:
The way to convert doc into byte array is using gson! All byte array are basically the JSON byte array

Reply:
So basically you dont need the code above at all.  

```java
// Code to convert an object to a string
String jsonInString = gson.toJson(obj);
```

Reply:
Done